core/backend/src/CloudStorageBackend.ts

```diff
@@ -33,7 +33,7 @@ export abstract class CloudStorageService {
   public terminate(): void { }
   public abstract id: CloudStorageProvider;
   public abstract obtainContainerUrl(id: CloudStorageContainerDescriptor, expiry: Date, clientIp?: string): CloudStorageContainerUrl;
-  public abstract upload(container: string, name: string, data: Uint8Array, options?: CloudStorageUploadOptions): Promise<string>;
+  public abstract upload(container: string, name: string, data: Uint8Array, metadata?: any, options?: CloudStorageUploadOptions): Promise<string>;

```

Comment:
Can we move this to the end?  Seems less likely to accidently break existing third-party implementations that way...